packages/cdk8s/src/api-object.ts

```diff
@@ -166,17 +165,31 @@ export class ApiObject extends Construct {
 
   /**
    * Renders the object to Kubernetes JSON.
+   *
+   * To disable sorting of dictionary keys in output object set the
+   * `CDK8S_DISABLE_SORT` environment variable.
    */
   public toJson(): any {
-    const data = {
+    const data: any = {
       ...this.props,
       metadata: this.metadata.toJson(),
     };
 
-    // convert to "pure data" so, for example, when we convert to yaml these
-    // references are not converted to anchors.
-    const json = JSON.parse(stringify(sanitizeValue(resolve(data))));
-    return JsonPatch.apply(json, ...this.patches);
+    const sortKeys = process.env.CDK8S_DISABLE_SORT ? false : true;
+    const json = sanitizeValue(resolve(data), { sortKeys });
+    const patched = JsonPatch.apply(json, ...this.patches);
+
+    // reorder top-level keys so that we first have "apiVersion", "kind" and
+    // "metadata" and then all the rest
+    const result: any = {};
+    const orderedKeys = ['apiVersion', 'kind', 'metadata', ...Object.keys(patched)];
+    for (const k of orderedKeys) {
+      if (k in patched) {
```

Comment:
When can this be `false`?

Reply:
Probably never, but just in case one of the primary fields was not set

Reply:
Wouldn't it be better to throw in that case? Seems like an illegal situation.

Reply:
Why illegal? There could be an object without metadata for example.

Reply:
Hmm. Can there? I thought these 3 are mandatory for all objects according to k8s spec.

Reply:
I think metadata is technically optional (and even apiVersion for things like `List` if I recall)

Reply:
Keep forgetting about those. ok.